File: 1.Elementary level/binary-search.py
def binary_search(list, item):
    low  = 0
    high = len(list) - 1
    while low <= high:
        mid   = (low + high)
        guess = list[mid]
        if guess == item:
            return mid
        if guess > item:
            high = mid - 1
        else:
            low  = mid + 1
    return None
l = [1,3,5,6,7,8,19]
print(binary_search(l,8))

File: 1.Elementary level/binary_search.js
function binarySearch(array,item){
    let low   = 0, 
    high      = array.length -1,
    mid,
    guess;
    while(low<=high){
        mid   = low+high
        guess = array[mid]
        if (guess===item){
            return mid
        }else if(guess<item){
            low = mid+1
        }else{
            high = mid-1
        }
    }
    return null
}
const arr = [1,3,5,6,7,8,19]
console.log(binarySearch(arr,8))
